pkg/relayer/miner/options.go

```diff
@@ -0,0 +1,11 @@
+package miner
+
+import "github.com/pokt-network/poktroll/pkg/relayer"
+
+// WithDifficulty sets the difficulty of the miner, where difficultyBytes is the
+// minimum number of leading zero bytes.
+func WithDifficulty(difficultyBits int) relayer.MinerOption {
```

Comment:
I think this is more of a question of which component should have the responsibility of querying that parameter. The `miner` is more re-usable and less complex if it is configured with the difficulty rather than querying for it itself. This also makes it more easily testable. I feel like it makes more sense for the `relayMiner` to encapsulate the querying of that configuration (or to delegate it to a third component).